### DISCUSSION TITLE
How to download podman for installation in non internet machine

### DISCUSSION THREAD

**Discussion by lchunleo** [containers/podman#21263] [Q&A]
Can I check how can I download podman in internet machine , and then transfer to non internet machine to.install? Thanks

**Comment by rhatdan** [comment]
What platform?  On Linux we use standard packaging, so the packageing has to be inside of the firewall. For Mac and Windows, I think you can download the package from podman.io

**Reply by rhatdan** [reply]
You can also grab the packages for Windows/Mac from this site

https://github.com/containers/podman/releases/

**Reply by lchunleo** [reply]
for linux platform. i had consulted the podman website that for ubuntu it is sudo apt-get -y install podman.

**Comment by kavishgr** [comment]
Static binaries for podman: [podman-static](https://github.com/mgoltzsche/podman-static)

Pretty cool project. It's just using docker to compile podman, all its dependencies, and spit out the binaries.

Just run this:

```
curl -fsSL -o podman-linux-amd64.tar.gz https://github.com/mgoltzsche/podman-static/releases/latest/download/podman-linux-amd64.tar.gz
```

You can extract the files in `$HOME/.podman/bin` and put the location in your `$PATH`. And you're good to go(assuming `newuidmap/newgidmap` is installed on your machine). I'm currently using it on my steam deck with `distrobox` and `toolbox`. 

There's also [podman-launcher](https://github.com/89luca89/podman-launcher), which uses the podman-static project to compile podman and all its dependencies into a **single** binary(haven't tried it yet though).

**Reply by rhatdan** [reply]
You need to realize that Podman requires or potentially needs a few other binaries to run.
crun
catatonit
conmon
containers-common-extra
gvisor-tap-vsock-gvforwarder
gvisor-tap-vsock
runc
crun
pasta 
slirp4netns
aardvark-dns
netavark

**Reply by kavishgr** [reply]
Sorry I forgot to mention this. Yes it does provide some of the dependencies you mentioned:

```
./local/bin:
fuse-overlayfs  fusermount3  podman  runc  slirp4netns

./local/lib:
cni  podman

./local/lib/cni:
bridge  firewall  host-local  loopback  portmap  tuning

./local/lib/podman:
catatonit  conmon  rootlessport
```

 It's not a full fledged podman. But it works without any issues for my case.

**Reply by KhazAkar** [reply]
I'd even propose pulling such releases onto official podman umbrella. It might help with keeping podman up-to-date on Ubuntu 22.04 and it's derivatives without resorting to KVM or manual compilation step.
